sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/ConfigProperties.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.autoconfigure;
+
+import java.util.AbstractMap;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+class ConfigProperties {
+
+  static ConfigProperties get() {
+    return new ConfigProperties(System.getProperties(), System.getenv());
+  }
+
+  // Visible for testing
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  static ConfigProperties createForTest(Map<String, String> properties) {
+    return new ConfigProperties((Map) properties, Collections.emptyMap());
+  }
+
+  private final Map<String, String> config;
```

Comment:
this should be up above the static methods

Reply:
Heh - I'm going to have trouble getting used to fields being so far from the constructor :P

Reply:
that's why static methods should go at the bottom of the class 😁